fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/SimplifyWindowExpression.java

```diff
@@ -0,0 +1,129 @@
+
+    @Override
+    public Rule build() {
+        return logicalWindow(any()).thenApply(this::simplify)
+                .toRule(RuleType.SIMPLIFY_WINDOW_EXPRESSION);
+    }
+
+    private Plan simplify(MatchingContext<LogicalWindow<Plan>> ctx) {
+        LogicalWindow<Plan> window = ctx.root;
+        ImmutableList.Builder<NamedExpression> projectionsBuilder = ImmutableList.builder();
+        ImmutableList.Builder<NamedExpression> remainWindowExpression = ImmutableList.builder();
+        List<NamedExpression> windowExpressions = window.getWindowExpressions();
+        for (NamedExpression expr : windowExpressions) {
+            Alias alias = (Alias) expr;
+            WindowExpression windowExpression = (WindowExpression) alias.child();
+            if (!windowExpression.getOrderKeys().isEmpty()) {
+                remainWindowExpression.add(expr);
+                continue;
+            }
+            if (windowExpression.getPartitionKeys().stream().anyMatch((
+                    partitionKey -> partitionKey.getDataType().isOnlyMetricType()))) {
+                continue;
+            }
+            // after normalize window, partition key must be slot
+            List<Slot> partitionSlots = (List<Slot>) (List) windowExpression.getPartitionKeys();
+            Set<Slot> partitionSlotSet = new HashSet<>(partitionSlots);
+            if (!window.getLogicalProperties().getFunctionalDependencies().isUnique(partitionSlotSet)) {
+                remainWindowExpression.add(expr);
+                continue;
+            }
+            Expression function = windowExpression.getFunction();
+            if (function instanceof BoundFunction) {
+                BoundFunction boundFunction = (BoundFunction) function;
+                String name = ((BoundFunction) function).getName();
+                if ((name.equals(COUNT) && boundFunction.child(0).notNullable())
+                        || REWRRITE_TO_CONST_WINDOW_FUNCTIONS.contains(name)) {
+                    projectionsBuilder.add(new Alias(alias.getExprId(), new TinyIntLiteral((byte) 1), alias.getName()));
+                } else if (REWRRITE_TO_SLOT_WINDOW_FUNCTIONS.contains(name)) {
+                    projectionsBuilder.add(new Alias(alias.getExprId(), boundFunction.child(0), alias.getName()));
+                } else {
+                    remainWindowExpression.add(expr);
+                }
+            } else {
+                remainWindowExpression.add(expr);
+            }
+        }
+        List<NamedExpression> projections = projectionsBuilder.build();
+        List<NamedExpression> remainWindow = remainWindowExpression.build();
+
+        if (projections.isEmpty()) {
+            return window;
+        } else if (remainWindow.isEmpty()) {
+            return new LogicalProject(projections, window.child(0));
+        } else {
+            return new LogicalProject(projections, window.withExpression(remainWindow,
+                    window.child(0)));
```

Comment:
do u have case to test this branch? i think this is not right because project should contain remain window node's output

Reply:
The problem has been modified and regression cases have been added